addons/disable-auto-save/userscript.js

```diff
@@ -1,13 +1,67 @@
 export default async ({ addon, console, msg }) => {
+  function appendChildWithAttributes(parent, type, attrs = {}) {
+    const element = document.createElement(type);
+    Object.assign(element, attrs);
+    parent.appendChild(element);
+    return element;
+  }
+  const PROJECT_PAGE_BUTTON = "[class*='community-button_community-button_']";
+
   addon.tab.redux.initialize();
   addon.tab.redux.addEventListener("statechanged", ({ detail }) => {
     if (addon.self.disabled || detail.action.type !== "timeout/SET_AUTOSAVE_TIMEOUT_ID") return;
     clearTimeout(detail.next.scratchGui.timeout.autoSaveTimeoutId);
     console.log("Pending autosave prevented.");
   });
 
+  function showSaveAlert() {
+    const modal = addon.tab.createModal(msg("alert-title"), {
+      isOpen: true,
+      useEditorClasses: true,
+    });
+    modal.container.classList.add("unsavedChangesDialog");
+
+    appendChildWithAttributes(modal.content, "p", { textContent: msg("unsaved-changes") });
+    const buttonRow = appendChildWithAttributes(modal.content, "div", {
+      className: addon.tab.scratchClass("prompt_button-row", { others: "unsavedChangesDialog-actions" }),
+    });
+    const buttons = {};
+    for (const option of ["cancel", "discard", "save"]) {
+      buttons[option] = appendChildWithAttributes(buttonRow, "button", {
+        value: option,
+        textContent: msg(option),
+      });
+    }
+    buttons.save.classList.add(addon.tab.scratchClass("prompt_ok-button"));
+
+    // On any button click:
+    buttonRow.addEventListener(
+      "click",
+      async (e) => {
+        if (!e.target.matches("button")) return;
+        // All buttons close the modal
+        modal.remove();
+
+        if (e.target.value === "discard") {
+          // Mark changes as saved to avoid a save-on-navigation
+          addon.tab.redux.dispatch({
+            type: "scratch-gui/project-changed/SET_PROJECT_CHANGED",
+            changed: false,
+          });
```

Comment:
Setting `changed` to false will disable the confirmation message that normally appears when closing the page. You should set it back to true after switching to the project page.